### PR TITLE
Issue/2475 : Tooltip descriptions in launcher for window configs

### PR DESCRIPTION
**Pull request by GPayne**
As with the profiles, this provides a tool tip description for each config option from the built-in configs. Appears when user hovers over the config filename in the launcher drop-down menu.

### CODE REVIEW

File: config/single_fisheye.json [hunk 1]
```diff
@@ -33,5 +33,12 @@
       "eyeseparation": 0.06,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with fisheye projection rendering.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_fisheye.json:41]:
```suggestion
    "name": "Fisheye",
```

File: config/single_fisheye_gui.json [hunk 2]
```diff
@@ -57,5 +57,13 @@
       "eyeseparation": 0.06,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows: Primary 1024x1024 window with menu and text overlays minus rendering, and a secondary 1024x1024 window with fisheye projection rendering without menu and text overlays.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_fisheye_gui.json:65]:
```suggestion
    "name": "Fisheye and GUI (no rendering)",
```

File: config/single_gui.json [hunk 3]
```diff
@@ -66,5 +66,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows: Primary 1280x720 window with planar projection rendering without menu or text overlays, and a secondary with menu and text overlays minus rendering.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_gui.json:74]:
```suggestion
    "name": "Planar and GUI (no rendering)",
```

File: config/single_gui_spout.json [hunk 4]
```diff
@@ -63,5 +63,12 @@
       "eyeseparation": 0.06499999761581421,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows: Primary 1280x720 window with Spout output (name 'OpenSpace') planar projection rendering without menu or text overlays, and a secondary with menu and text overlays minus rendering.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_gui_spout.json:71]:
```suggestion
    "name": "Spout and GUI (no rendering)",
```

File: config/single_gui_with_graphics.json [hunk 5]
```diff
@@ -67,5 +67,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows: Primary 1280x720 window with planar projection rendering without menu or text overlays, and a secondary with menu and text overlays plus rendering from the primary.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_gui_with_graphics.json:75]:
```suggestion
    "name": "Planar and GUI (with rendering)",
```

Comment (WeirdRubberDuck) [reply config/single_gui_with_graphics.json:75]: Naming is hard... Feel free to come up with better names

File: config/single_sbs_stereo.json [hunk 6]
```diff
@@ -42,5 +42,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1280x720 window containing side-by-side stereo planar projection rendering views.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_sbs_stereo.json:50]:
```suggestion
    "name": "Side-by-side Stereo (planar)",
```

File: config/equirectangular_gui.json [hunk 7]
```diff
@@ -54,5 +54,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows: Primary 1280x720 window with equirectangular projection rendering without menu or text overlays, and a secondary with menu and text overlays minus rendering.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/equirectangular_gui.json:62]:
```suggestion
    "name": "Equirectangular and GUI",
```

File: config/fullscreen1080.json [hunk 8]
```diff
@@ -43,5 +43,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1920x1080 fullscreen window with planar projection rendering. The display will automatically be set to 1920x1080, regardless of native resolution.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/fullscreen1080.json:51]:
```suggestion
    "name": "1080p Fullscreen",
```

File: config/gui_projector.json [hunk 9]
```diff
@@ -59,5 +59,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows for a side-by-side dual-monitor configuration, where the right-side monitor can be considered a projector. Primary 1920x1080 window with menu and text overlays minus rendering, and a secondary 1920x1080 fullscreen window with planar projection rendering without menu or text overlays.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/gui_projector.json:67]:
```suggestion
    "name": "GUI and Projector",
```

File: config/single.json [hunk 10]
```diff
@@ -43,5 +43,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1280x720 window with planar projection rendering at 2560x1440 resolution.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single.json:51]: Another name? Should be more descriptive than just "Single". 

I tried removing the Single Node aspect in my other suggestions. Thoughts?

File: config/single_two_win.json [hunk 11]
```diff
@@ -55,5 +55,13 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two 1280x720 windows each with identical planar projection rendering. Only the primary window has menu controls, but both have text overlays.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/single_two_win.json:63]:
```suggestion
    "name": "Two Windows",
```

File: config/spherical_mirror.json [hunk 12]
```diff
@@ -36,5 +36,12 @@
       "eyeseparation": 0.06,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1280x720 window with spherical projection rendering at 2048x2048 resolution.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/spherical_mirror.json:44]:
```suggestion
    "name": "Spherical Mirror Projection",
```

File: config/spherical_mirror_gui.json [hunk 13]
```diff
@@ -61,5 +61,12 @@
       "eyeseparation": 0.06,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two windows: Primary 1280x720 window with menu and text overlays minus rendering, and a secondary 1280x720 window at 2048x2048 resolution with spherical mirror projection rendering without the menu or text overlays.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/spherical_mirror_gui.json:69]:
```suggestion
    "name": "Spherical Mirror Projection and GUI (no rendering)",
```

File: config/spout_output_cubemap.json [hunk 14]
```diff
@@ -56,5 +56,12 @@
       "eyeseparation": 0.05999999865889549,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with spout output cubemap projection rendering.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/spout_output_cubemap.json:64]:
```suggestion
    "name": "Spout Output - Cubemap",
```

File: config/spout_output_cubemap.json [hunk 15]
```diff
@@ -56,5 +56,12 @@
       "eyeseparation": 0.05999999865889549,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with spout output cubemap projection rendering.",
```
Comment (WeirdRubberDuck) [inline config/spout_output_cubemap.json:62]:
```suggestion
    "description": "A single 1024x1024 window with Spout output cubemap projection rendering.",
```

File: config/spout_output_equirectangular.json [hunk 16]
```diff
@@ -56,5 +56,12 @@
       "eyeseparation": 0.06,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with spout output equirectangular projection rendering.",
```
Comment (WeirdRubberDuck) [inline config/spout_output_equirectangular.json:62]:
```suggestion
    "description": "A single 1024x1024 window with Spout output equirectangular projection rendering.",
```

File: config/spout_output_equirectangular.json [hunk 17]
```diff
@@ -56,5 +56,12 @@
...
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/spout_output_equirectangular.json:64]:
```suggestion
    "name": "Spout Output - Equirectangular",
```

File: config/spout_output_fisheye.json [hunk 18]
```diff
@@ -56,5 +56,12 @@
       "eyeseparation": 0.06,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with spout output fisheye projection rendering.",
```
Comment (WeirdRubberDuck) [inline config/spout_output_fisheye.json:62]:
```suggestion
    "description": "A single 1024x1024 window with Spout output fisheye projection rendering.",
```

File: config/spout_output_fisheye.json [hunk 19]
```diff
@@ -56,5 +56,12 @@
...
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/spout_output_fisheye.json:64]:
```suggestion
    "name": "Spout Output - Fisheye",
```

File: config/spout_output_flat.json [hunk 20]
```diff
@@ -55,5 +55,12 @@
       "eyeseparation": 0.05999999865889549,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with spout output planar/flat projection rendering at 1920x1080 resolution.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/spout_output_flat.json:63]:
```suggestion
    "name": "Spout Output - Flat",
```

File: config/spout_output_flat.json [hunk 21]
```diff
@@ -55,5 +55,12 @@
       "eyeseparation": 0.05999999865889549,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "A single 1024x1024 window with spout output planar/flat projection rendering at 1920x1080 resolution.",
```
Comment (WeirdRubberDuck) [inline config/spout_output_flat.json:61]:
```suggestion
    "description": "A single 1024x1024 window with Spout output planar/flat projection rendering at 1920x1080 resolution.",
```

File: config/two_nodes.json [hunk 22]
```diff
@@ -60,5 +60,12 @@
       "eyeseparation": 0.065,
       "pos": { "x": 0.0, "y": 0.0, "z": 0.0 }
     }
-  ]
+  ],
+  "meta": {
+    "author": "OpenSpace Team",
+    "description": "Two individual nodes, each with a 1280x720 window with planar projection rendering.",
+    "license": "MIT License",
+    "name": "Single",
```
Comment (WeirdRubberDuck) [inline config/two_nodes.json:68]:
```suggestion
    "name": "Two Nodes",
```

File: apps/OpenSpace/ext/launcher/src/launcherwindow.cpp [hunk 23]
```diff
@@ -552,6 +552,17 @@ bool handleConfigurationFile(QComboBox& box, const std::filesystem::directory_en
     }
     box.addItem(QString::fromStdString(p.path().filename().string()));
 
+    //Add tooltip
```
Comment (WeirdRubberDuck) [inline apps/OpenSpace/ext/launcher/src/launcherwindow.cpp:555]:
```suggestion
    // Add tooltip
```

**Review by WeirdRubberDuck — APPROVED** [review]
Looks good to me. Note that the name field of all configs has to be set though (even though I know it's not really used by now)

I gave some suggestions and tried avoiding to use the word "Single" in the name, as almost all our files are targeted towards a single node. Let me know what you think and feel free to overwrite with your own suggestions.

**Comment by WeirdRubberDuck** [comment]
I guess the next step is to add the possibility to add that meta information in the Configuration Editor, when creating/editing a file? :)

**Comment by GPayne** [comment]
I made the name fixes requested. I opted for brevity in the naming since the descriptions provide all the detail.

**Pull request merged**